src/vs/base/parts/quickinput/browser/quickInputBox.ts

```diff
@@ -91,7 +91,15 @@ export class QuickInputBox extends Disposable {
 	}
 
 	set enabled(enabled: boolean) {
-		this.findInput.setEnabled(enabled);
+		// We can't disable the input box because it is still used for
+		// navigating the list. Instead, we disable the list and the OK
+		// so that nothing can be selected.
+		// TODO: should this be what we do for all find inputs? Or maybe some _other_ API
+		// on findInput to change it to readonly?
+		this.findInput.inputBox.inputElement.toggleAttribute('readonly', !enabled);
+		// TODO: styles of the quick pick need to be moved to the CSS instead of being in line
+		// so things like this can be done in CSS
+		// this.findInput.inputBox.inputElement.classList.toggle('disabled', !enabled);
```

Comment:
Feels bad leaving this as readonly, if this is just to preserve focus could we do `once('blur', () => clearReadonlyAndSetDisabled())`?

Reply:
talked offline. This isn't too much of a concern but I do ack that there's some overloading going on with enable, disable, and readonly.